stm32-modules/heater-shaker/include/heater-shaker/gcodes.hpp

```diff
@@ -119,29 +120,33 @@ struct GetTemperature {
     requires std::forward_iterator<InputIt>&&
         std::sized_sentinel_for<InputIt, InLimit> static auto
         write_response_into(InputIt buf, InLimit limit,
-                            uint32_t current_temperature,
-                            uint32_t setpoint_temperature) -> InputIt {
+                            double current_temperature,
+                            double setpoint_temperature) -> InputIt {
         static constexpr const char* prefix = "M105 C";
```

Comment:
Pardon the dumb questions....

If you're using snprintf, why not go full on with the format string `"M105 C%02.F T%02.F"`?

What the heck is all this code in the body of a `struct`?

Reply:
that's... a good point about the full format string. i probably should do this, yeah.

as to the struct thing - c++ structs are (more or less) classes with default `public` visibility, so these are (static) member functions.

Reply:
Got it. Thanks